### ISSUE TITLE
Notification "archive all" button not applied to all notifications

### ISSUE THREAD

**Issue by semorur** [directus/directus#22265]
### Describe the Bug

Notification "archive all" button not working

### To Reproduce

when you have any of notification and you press "archive all" button. it doesn't work. nothing happens

<img width="850" alt="1" src="https://github.com/directus/directus/assets/152224102/777a28e0-6297-4906-9d30-fd7a8b1fae18">

I have to tick manually all buttons and press button "archive" to make it

<img width="855" alt="2" src="https://github.com/directus/directus/assets/152224102/0e29bf3e-cc10-476f-983f-0485698630c2">

### Directus Version

v10.10.5

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by hanneskuettner** [comment]
I've tried reproducing this on my end both in v10.10.5 and the newest release v10.10.7 and am unable to see any problem.

Can you check your browser console for any errors, as well as verify that a request to `PATCH /notifications` is made?

**Comment by semorur** [comment]
Hello

I updated to v10.10.7 and have the same issue
tried from Safari and Chrome

{
  "query": {
    "recipient": {
      "_eq": "f652e6b7-c80e-4bf2-b9e7-9cbce371ca79"
    }
  },
  "data": {
    "status": "archived"
  }
}


Status code 200


in preview and response there are only 100 items
first 100 id, but my notifications are not in this list

Response gives me not only my notifications, but also from other users.

**Comment by DuckThom** [comment]
I'm having the same issue where the "Archive All" button is not working properly. 

Though for me the PATCH request does produce a 200 response `directus-1  | [11:09:15] PATCH /notifications 200 592ms` but no notifications are cleared. In my case it's trying to archive thousands of notifications.

I've also tried removing them through a Flow but the delete data operation seems to try to load all the records first which seems to crash Directus. (API unavailable message when running the Flow, followed by being redirected to the login screen)

**Comment by paescuj** [comment]
The permission issue for non-admin users has now been fixed via #23411, which improves the situation. Though, the issue with the query limit is still present.
My preferred solution for this would be the third option I've suggested in https://github.com/directus/directus/pull/23411#pullrequestreview-2245604875
> Replace the current "Archive All" button with a "Select All" option as suggested in https://github.com/directus/directus/pull/19503#issuecomment-1692586807

@Dominic-Marcelino 
> Would also feel right for me, but would be one click extra related to the current solution. Wdyt in relation to the first thoughts?

I wouldn't mind the extra click, as on the one hand it would be consistent with the other layouts and on the other hand would clearly indicate to the user what happens when the "Archive" button is clicked (not that clear with the "Archive All" button - does it archive all items on current page vs. really all).
